src/cudamatrix/cu-allocator.cc

```diff
@@ -185,6 +185,19 @@ void* CuMemoryAllocator::MallocFromSubregion(SubRegion *subregion,
   // region was sufficiently large.  We don't check this; if it segfaults, we'll
   // debug.
 
+  // search for a block that we don't have to synchronize on
+  int max_iters = 20;
+  auto chosen_iter = iter;
```

Comment:
oh, sorry, regarding the naming I just realized that I misinterpreted the code.  Can you please change chosen_iter to iter2?

Reply:
How about `search_iter`?